Updates

tiwarishrijan opened the pull request:
Updates

tiwarishrijan committed:
added session restore function

We can pickle session details to store it in database or file and reload each time for new connection..

tiwarishrijan committed:
To connection status

tiwarishrijan committed:
Send Text Message

1 > Added function sendTextMessage to send text message.
2 > Added function to_bytes
3 > Changed Whatsapp Encrypt function to return Hash+ciphertext instead of ciphertext+sign

tiwarishrijan committed:
Updated MessageID for text Message

Message ID starts with 3EB0_____________________

sigalor committed:
Merge pull request #122 from tiwarishrijan/patch-1

added session restore function

sigalor committed:
Merge branch 'master' into patch-2

sigalor committed:
Merge pull request #123 from tiwarishrijan/patch-2

To connection status

sigalor committed:
Merge branch 'master' into patch-4

sigalor committed:
Merge pull request #124 from tiwarishrijan/patch-4

Send Text Message

tiwarishrijan committed:
Keep Alive

sigalor committed:
Merge pull request #127 from tiwarishrijan/patch-3

Keep Alive

quad committed:
sass: ruby -> node

tiwarishrijan committed:
Updated Media Upload

- fieldname: `file`, filename: `blob`: should contain enc+mac as a value

sigalor committed:
Merge pull request #131 from quad/sass

Switch from deprecated ruby sass to node sass

sigalor committed:
Merge pull request #132 from tiwarishrijan/patch-5

Updated Media Upload

sigalor committed:
Add protobuf-extractor

sigalor committed:
Remove --no-source-map from Sass command

sigalor committed:
Update protobuf definitions

sigalor committed:
Use WhatsApp Web version 0.3.1649

sigalor committed:
Add syntax and package declaration to protobuf definitions

sigalor committed:
Update WhatsApp Web version to 0.3.2390

sigalor committed:
Update all Node dependencies

sigalor committed:
Update all dependencies of the protobuf-extractor subproject

sigalor committed:
Update protobuf definitions as of April 6, 2019

sigalor committed:
Convert tabs to spaces in Python code

sigalor committed:
Convert tabs to spaces in frontend code

sigalor committed:
Convert tabs to spaces in remaining code

sigalor committed:
Add new metrics to WhatsApp defines

sigalor committed:
Fix indentation in whatsapp.py (#178)

abdurrahmanekr committed:
Collapse/Expand Feature Added.

tulir committed:
Update protobuf extractor to work on latest whatsapp web

tulir committed:
Fix inline enums in protobuf extractor

tulir committed:
Sort modules so protobuf schema stays in same order

tulir committed:
Stop creating multiple protobuf headers

tulir committed:
Update protobuf definitions

Commit:
Change endpoint

Makhlab committed:
change whatsapp websocket url in index_jsdemo.js

dependabot[bot] committed:
Bump mixin-deep from 1.3.1 to 1.3.2

Bumps [mixin-deep](https://github.com/jonschlinkert/mixin-deep) from 1.3.1 to 1.3.2.
- [Release notes](https://github.com/jonschlinkert/mixin-deep/releases)
- [Commits](https://github.com/jonschlinkert/mixin-deep/compare/1.3.1...1.3.2)

Signed-off-by: dependabot[bot] <support@github.com>

quad committed:
Merge pull request #233 from sigalor/dependabot/npm_and_yarn/mixin-deep-1.3.2

Bump mixin-deep from 1.3.1 to 1.3.2

dependabot[bot] committed:
Bump lodash from 4.17.11 to 4.17.13

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.11 to 4.17.13.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.11...4.17.13)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump lodash from 4.17.11 to 4.17.15 in /doc/spec/protobuf-extractor

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.11 to 4.17.15.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.11...4.17.15)

Signed-off-by: dependabot[bot] <support@github.com>

quad committed:
Merge pull request #235 from sigalor/dependabot/npm_and_yarn/doc/spec/protobuf-extractor/lodash-4.17.15

Bump lodash from 4.17.11 to 4.17.15 in /doc/spec/protobuf-extractor

quad committed:
Merge pull request #234 from sigalor/dependabot/npm_and_yarn/lodash-4.17.13

Bump lodash from 4.17.11 to 4.17.13

dependabot[bot] committed:
Bump lodash from 4.17.13 to 4.17.15

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.13 to 4.17.15.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.13...4.17.15)

Signed-off-by: dependabot[bot] <support@github.com>

quad committed:
Merge pull request #236 from sigalor/dependabot/npm_and_yarn/lodash-4.17.15

Bump lodash from 4.17.13 to 4.17.15

quad committed:
Change endpoint (merge PRs #221 and #222)

Thanks to @David-Kurniawan and @Makhlab

quad committed:
Merge pull request #195 from abdurrahmanekr/collapse-expand

Collapse/Expand Feature Added.

quad committed:
Merge pull request #220 from tulir/fix-protobuf-extractor

Fix protobuf extractor

elonsalfati committed:
added new tokens

J4YF7O committed:
Remove .vscode directory

J4YF7O committed:
Remove idea directory (for JetBrains IDEs)

J4YF7O committed:
Generate yarn.lock file using yarn import

J4YF7O committed:
Create script for docker command

J4YF7O committed:
Create new dockerfile based on node lastest image

J4YF7O committed:
Chapter on docker added.

J4YF7O committed:
Image name for docker run added. #Fix

J4YF7O committed:
Fix bug on issue with contacts #158

J4YF7O committed:
Fix bug on front-end port (2018 instead 2019) for docker

quad committed:
Merge pull request #255 from J4YF7O/feature/user-contacts-fix

Fix : issue with contacts

quad committed:
Merge pull request #256 from J4YF7O/feature/ide-gitignore

Feature/ide gitignore

quad committed:
Merge pull request #257 from J4YF7O/feature/yarn-support

Add yarn support

quad committed:
Merge pull request #258 from J4YF7O/feature/docker-support

Add docker support

quad committed:
Merge pull request #254 from ElonSalfati/tokens

added new tokens

sigalor committed:
Update dependencies

dependabot[bot] committed:
Bump acorn from 6.1.1 to 6.4.1 in /doc/spec/protobuf-extractor

Bumps [acorn](https://github.com/acornjs/acorn) from 6.1.1 to 6.4.1.
- [Release notes](https://github.com/acornjs/acorn/releases)
- [Commits](https://github.com/acornjs/acorn/compare/6.1.1...6.4.1)

Signed-off-by: dependabot[bot] <support@github.com>

quad committed:
Merge pull request #264 from sigalor/dependabot/npm_and_yarn/doc/spec/protobuf-extractor/acorn-6.4.1

Bump acorn from 6.1.1 to 6.4.1 in /doc/spec/protobuf-extractor

tulir committed:
Update protobuf extractor to work on latest whatsapp web

tulir committed:
Update protobuf definitions

quad committed:
Merge pull request #265 from tulir/master

Fix protobuf extractor (again)

alvarofe committed:
Update version of whatsapp web

quad committed:
Merge pull request #268 from alvarofe/dev

Update version of whatsapp web

joBr99 committed:
Run apt-get update & apt-get install in one layer

Each RUN command in a Dockerfile add's another layer, running update and install in different layers could cause issues, because the docker layers are cached. According to dockerfile's best practices this two command sould never be run in two statments.

See RUN/APT-GET in: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/

LaloHao committed:
Add Nix integration

adiwajshing committed:
Update README.md

ndunks committed:
Update README.md

Add other JS Reimplementations

quad committed:
Merge pull request #285 from ndunks/patch-1

Update README.md

quad committed:
Merge pull request #281 from joBr99/master

Run apt-get update & apt-get install in one layer

dependabot[bot] committed:
Bump lodash from 4.17.15 to 4.17.19

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.19.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.19)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump lodash from 4.17.15 to 4.17.19 in /doc/spec/protobuf-extractor

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.19.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.19)

Signed-off-by: dependabot[bot] <support@github.com>

quad committed:
Merge pull request #295 from sigalor/dependabot/npm_and_yarn/doc/spec/protobuf-extractor/lodash-4.17.19

Bump lodash from 4.17.15 to 4.17.19 in /doc/spec/protobuf-extractor

quad committed:
Merge pull request #294 from sigalor/dependabot/npm_and_yarn/lodash-4.17.19

Bump lodash from 4.17.15 to 4.17.19

dependabot[bot] committed:
Bump tree-kill from 1.2.0 to 1.2.2

Bumps [tree-kill](https://github.com/pkrumins/node-tree-kill) from 1.2.0 to 1.2.2.
- [Release notes](https://github.com/pkrumins/node-tree-kill/releases)
- [Commits](https://github.com/pkrumins/node-tree-kill/compare/v1.2.0...v1.2.2)

Signed-off-by: dependabot[bot] <support@github.com>

quad committed:
Merge pull request #297 from sigalor/dependabot/npm_and_yarn/tree-kill-1.2.2

Bump tree-kill from 1.2.0 to 1.2.2

dependabot[bot] committed:
Bump ini from 1.3.5 to 1.3.8

Bumps [ini](https://github.com/isaacs/ini) from 1.3.5 to 1.3.8.
- [Release notes](https://github.com/isaacs/ini/releases)
- [Commits](https://github.com/isaacs/ini/compare/v1.3.5...v1.3.8)

Signed-off-by: dependabot[bot] <support@github.com>

AvTe commented:
Encrypt function to return Hash+ciphertext instead of ciphertext+sign 

backend/whatsapp.py 
dir

dependabot[bot] committed:
Bump lodash from 4.17.19 to 4.17.21

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.19 to 4.17.21.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.19...4.17.21)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump lodash from 4.17.19 to 4.17.21 in /doc/spec/protobuf-extractor

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.19 to 4.17.21.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.19...4.17.21)

Signed-off-by: dependabot[bot] <support@github.com>

quad committed:
Merge pull request #329 from sigalor/dependabot/npm_and_yarn/doc/spec/protobuf-extractor/lodash-4.17.21

Bump lodash from 4.17.19 to 4.17.21 in /doc/spec/protobuf-extractor

quad committed:
Merge pull request #328 from sigalor/dependabot/npm_and_yarn/lodash-4.17.21

Bump lodash from 4.17.19 to 4.17.21

dependabot[bot] committed:
Bump dot-prop from 4.2.0 to 4.2.1

Bumps [dot-prop](https://github.com/sindresorhus/dot-prop) from 4.2.0 to 4.2.1.
- [Release notes](https://github.com/sindresorhus/dot-prop/releases)
- [Commits](https://github.com/sindresorhus/dot-prop/compare/v4.2.0...v4.2.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump hosted-git-info from 2.7.1 to 2.8.9

Bumps [hosted-git-info](https://github.com/npm/hosted-git-info) from 2.7.1 to 2.8.9.
- [Release notes](https://github.com/npm/hosted-git-info/releases)
- [Changelog](https://github.com/npm/hosted-git-info/blob/v2.8.9/CHANGELOG.md)
- [Commits](https://github.com/npm/hosted-git-info/compare/v2.7.1...v2.8.9)

Signed-off-by: dependabot[bot] <support@github.com>

quad committed:
Merge pull request #330 from sigalor/dependabot/npm_and_yarn/dot-prop-4.2.1

Bump dot-prop from 4.2.0 to 4.2.1

quad committed:
Merge pull request #331 from sigalor/dependabot/npm_and_yarn/hosted-git-info-2.8.9

Bump hosted-git-info from 2.7.1 to 2.8.9

quad committed:
Merge pull request #309 from sigalor/dependabot/npm_and_yarn/ini-1.3.8

Bump ini from 1.3.5 to 1.3.8

dependabot[bot] committed:
Bump tar from 4.4.8 to 4.4.15

Bumps [tar](https://github.com/npm/node-tar) from 4.4.8 to 4.4.15.
- [Release notes](https://github.com/npm/node-tar/releases)
- [Changelog](https://github.com/npm/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-tar/compare/v4.4.8...v4.4.15)

---
updated-dependencies:
- dependency-name: tar
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

hossinasaadi committed:
update WHATSAPP_WEB_VERSION 

if use the last versions, you can’t connect to Whatsapp

quad committed:
Merge pull request #336 from hossinasaadi/patch-1

Update WHATSAPP_WEB_VERSION, otherwise WhatsApp rejects the connection.

quad committed:
Merge pull request #335 from sigalor/dependabot/npm_and_yarn/tar-4.4.15

Bump tar from 4.4.8 to 4.4.15

quad committed:
Merge pull request #282 from LaloHao/feature/nix-integration

Add Nix integration

dependabot[bot] committed:
Bump ws from 3.3.3 to 5.2.3

Bumps [ws](https://github.com/websockets/ws) from 3.3.3 to 5.2.3.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/compare/3.3.3...5.2.3)

---
updated-dependencies:
- dependency-name: ws
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

quad committed:
Merge pull request #337 from sigalor/dependabot/npm_and_yarn/ws-5.2.3

Bump ws from 3.3.3 to 5.2.3

hossinasaadi committed:
restore session implement (#325)

* fix keep Alive Request
* restore session implement
* add client restore button

Co-authored-by: hossinasaadi <Hossin277>

nikneym committed:
Make index_jsdemo.js stable (#340)

* remove unnecessary require calls
* update whatsapp version for QR init message

dependabot[bot] committed:
Bump tar from 4.4.15 to 4.4.19 (#341)

Bumps [tar](https://github.com/npm/node-tar) from 4.4.15 to 4.4.19.
- [Release notes](https://github.com/npm/node-tar/releases)
- [Changelog](https://github.com/npm/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-tar/compare/v4.4.15...v4.4.19)

---
updated-dependencies:
- dependency-name: tar
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

OmerFI committed:
Update requirements.txt (#344)

Changed from `pycrypto` to `pycryptodome`.
The PyCrypto project is not maintained since the last commit on 23 Jun 2014. Instead of it, the PyCryptodome project (the fork of it) is maintained and is more stable.

vadolasi committed:
Add "kyros" to reimplementations list (#349)

dependabot[bot] committed:
Bump undefsafe from 2.0.2 to 2.0.5 (#355)

Bumps [undefsafe](https://github.com/remy/undefsafe) from 2.0.2 to 2.0.5.
- [Release notes](https://github.com/remy/undefsafe/releases)
- [Commits](https://github.com/remy/undefsafe/compare/v2.0.2...v2.0.5)

---
updated-dependencies:
- dependency-name: undefsafe
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ajv from 6.6.2 to 6.12.6 in /doc/spec/protobuf-extractor (#356)

Bumps [ajv](https://github.com/ajv-validator/ajv) from 6.6.2 to 6.12.6.
- [Release notes](https://github.com/ajv-validator/ajv/releases)
- [Commits](https://github.com/ajv-validator/ajv/compare/v6.6.2...v6.12.6)

---
updated-dependencies:
- dependency-name: ajv
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

purpshell committed:
chore(README): Update Baileys's URL (#389)
